### PR TITLE
Feat/get user locked total

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
- added zcncore.GetUserLockedTotal

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- zboxcli: https://github.com/0chain/zboxcli/pull/337

### CODE REVIEW

**Comment by dabasov** [comment]
@cnlangzi I'm surprised that [this one](https://github.com/0chain/gosdk/pull/580) is not merged yet , it also have the similar changes, can we merge current PR after [580](https://github.com/0chain/gosdk/pull/580) is merged?

**Comment by cnlangzi** [comment]
> @cnlangzi I'm surprised that [this one](https://github.com/0chain/gosdk/pull/580) is not merged yet , it also have the similar changes, can we merge current PR after [580](https://github.com/0chain/gosdk/pull/580) is merged?

We have to to waiting for any code owner to approve it. I am not sure why we need code owner settings on repo.

File: zcncore/wallet_base.go [hunk 1]
```diff
@@ -1009,6 +1013,42 @@ func GetStakePoolUserInfo(clientID string, cb GetInfoCallback) (err error) {
...
+	if err != nil {
+		return 0, errors.Throw(err, "invalid json format")
+	}
+
+	total, ok := result["total"]
+	if ok {
+		return total, nil
+	}
+
+	return 0, errors.New("", "invalid result")
```
Comment (peterlimg) [inline zcncore/wallet_base.go]: could be simple standard error `errors.New("invalid request")

File: zcncore/wallet_base.go [hunk 2]
```diff
@@ -1130,3 +1170,40 @@ func GetPublicEncryptionKey(mnemonic string) (string, error) {
...
+	cb.callback <- true
+}
+
+func (cb *getInfoCallback) Wait() (string, error) {
+	<-cb.callback
+	if cb.err == "" {
+		return cb.info, nil
+	}
+
+	return "", errors.New("", cb.err)
```
Comment (peterlimg) [inline zcncore/wallet_base.go]: Same as above

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: zcncore/wallet_base.go (hunk 1 above)
Comment (cnlangzi) [reply zcncore/wallet_base.go]: @peterlimg  fixed

File: zcncore/wallet_base.go (hunk 2 above)
Comment (cnlangzi) [reply zcncore/wallet_base.go]: @peterlimg  fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/get_user_locked_total  |
| zboxcli  | feat/get_user_locked_total  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: zboxcore/sdk/sdk.go [hunk 3]
```diff
@@ -954,6 +954,9 @@ func GetAllocationBlobbers(
 	params := make(map[string]string)
 	params["allocation_data"] = string(allocationData)
 
+	b, _ := json.Marshal(params)
+	fmt.Println(string(b))
```
Comment (peterlimg) [inline zboxcore/sdk/sdk.go]: Let's remove debug logs

File: zcncore/wallet_base.go [hunk 4]
```diff
@@ -13,6 +13,8 @@ import (
 	"strings"
 	"time"
 
+	gErrors "errors"
```
Comment (peterlimg) [inline zcncore/wallet_base.go]: Let's use `stdErrors` for errors package from standard lib

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: zboxcore/sdk/sdk.go (hunk 3 above)
Comment (cnlangzi) [reply zboxcore/sdk/sdk.go]: @peterlimg  fixed

File: zcncore/wallet_base.go (hunk 4 above)
Comment (cnlangzi) [reply zcncore/wallet_base.go]: @peterlimg  fixed

**Comment by peterlimg** [comment]
@cnlangzi let's merge #718 first, then resolve conflicts on this one.

**Comment by cnlangzi** [comment]
> @cnlangzi let's merge #718 first, then resolve conflicts on this one.

@peterlimg  rebased and fixed it

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | feat/get_user_locked_total  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**